app/workers/commit_monitor_handlers/commit/bugzilla_common/bugzilla_common.rb

```diff
@@ -0,0 +1,33 @@
+module CommitMonitorHandlers
+  module Commit
+    module BugzillaCommon
+      def product
+        Settings.commit_monitor.bugzilla_product
+      end
+
+      def with_bug(bug_id)
+        return unless block_given?
+        MiqToolsServices::Bugzilla.call do
+          output = ActiveBugzilla::Bug.find(:product => product, :id => bug_id)
+          if output.empty?
+            logger.error "Unable to write for bug id #{bug_id}: Not a '#{product}' bug."
+          else
+            bug = output.first
+            yield(bug)
+          end
+        end
+      rescue => err
+        logger.error "Unable to write for bug id #{bug_id}: #{err}"
+      end
+
+      def branch_valid?
+        if @branch.pull_request?
+          return true if self.class.handled_branch_modes.include?(:pr)
+        else
+          return true if self.class.handled_branch_modes.include?(:regular)
+        end
+        false
```

Comment:
I think that logic is incorrect for the case when the branch is a PR and the only handled mode is regular.  This seems like something I should just write a spec for.  I'm not that comfortable with rspec yet though ... Maybe @imtayadeway can help me out with this when he has some time.